src/NServiceBus.AcceptanceTests/DelayedDelivery/TimeoutManager/When_timeout_dispatch_fails.cs

```diff
@@ -0,0 +1,124 @@
+                        options.RouteToLocalEndpointInstance();
+                        return bus.Send(new MyMessage(), options);
+                    }))
+                    .WithEndpoint<ErrorSpy>()
+                    .Done(c => c.FailedTimeoutMovedToError)
+                    .Repeat(r => r.For<AllTransportsWithoutNativeDeferral>())
+                    .Should(c => Assert.AreEqual(5, c.NumTimesStorageCalled))
+                    .Run();
+        }
+
+        public class Context : ScenarioContext
+        {
+            public bool FailedTimeoutMovedToError { get; set; }
+            public int NumTimesStorageCalled { get; set; }
+        }
+
+        public class Endpoint : EndpointConfigurationBuilder
+        {
+            public Endpoint()
+            {
+                EndpointSetup<DefaultServer>(config =>
+                {
+                    config.EnableFeature<TimeoutManager>();
+                    config.UsePersistence<FakeTimeoutPersistence>();
+                    config.SendFailedMessagesTo(ErrorQueueForTimeoutErrors);
+                    config.RegisterComponents(c => c.ConfigureComponent<FakeTimeoutStorage>(DependencyLifecycle.SingleInstance));
+                });
+            }
+
+            class FakeTimeoutPersistence : PersistenceDefinition
+            {
+                public FakeTimeoutPersistence()
+                {
+                    Supports<StorageType.Timeouts>(s => { });
+                }
+            }
+
+            class FakeTimeoutStorage : IQueryTimeouts, IPersistTimeouts
+            {
+                public Context TestContext { get; set; }
+                
+                public Task<TimeoutsChunk> GetNextChunk(DateTime startSlice)
+                {
+                    var timeout = new TimeoutsChunk.Timeout(Guid.NewGuid().ToString(), DateTime.UtcNow);
+                    var timeouts = new List<TimeoutsChunk.Timeout>() { timeout };
+                    return Task.FromResult(new TimeoutsChunk(timeouts, DateTime.UtcNow + TimeSpan.FromSeconds(10)));
+                }
+
+                public Task Add(TimeoutData timeout, ContextBag context)
+                {
+                    return Task.FromResult(0);
+                }
+
+                public Task<bool> TryRemove(string timeoutId, ContextBag context)
+                {
+                    throw new NotImplementedException();
+                }
+
+                public Task<TimeoutData> Peek(string timeoutId, ContextBag context)
+                {
+                    TestContext.NumTimesStorageCalled++;
+                    throw new Exception("Ooops. Timeout storage went bust");
+                }
+
+                public Task RemoveTimeoutBy(Guid sagaId, ContextBag context)
+                {
+                    throw new NotImplementedException();
+                }
+            }
+        }
+
+        public class ErrorSpy : EndpointConfigurationBuilder
```

Comment:
We discussed this with @timbussmann and we acknowledge that `ErrorSpy` is a hack but it is the best way we currently know of that can reliably test behaviours that rely on message being dispatched in a specific way.

I refactored the tests to use exception asserts but an experiment with injecting faults to the behaviour under test clearly showed these injected bugs went undetected.

The ideal solution would be to run these tests with a development transport that we could assert on after the test is done without the need to spin up an artificial endpoint just to check if messages were properly dispatched.